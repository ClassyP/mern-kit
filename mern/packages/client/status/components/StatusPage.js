import Page from 'common/components/Page';
import React, {Component} from 'react';
import Paper from 'material-ui/Paper';

export default class StatusPage extends Component {

  render () {
    const title = window.statusTitle || document.title || '';
    const message = window.statusMessage || '';

    return (
      <Page>
        <Paper className='status__container' zDepth={3}>
          <h1 className='status-title'>{ title }</h1>
          <h3 className='status-message'>{ message }</h3>
        </Paper>
      </Page>
    );
  }
}
