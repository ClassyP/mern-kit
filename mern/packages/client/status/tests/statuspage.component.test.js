import StatusPage from 'status/components/StatusPage';
import renderer from 'react-test-renderer';
import {mount} from 'enzyme';
import React from 'react';

describe('StatusPage Component Test', () => {

  test('Mounting and Unmounting the StatusPage component should not throw', () => {
    const wrapper = mount(<StatusPage />);
    wrapper.unmount();
  });

  test('Snapshot test of the StatusPage component', () => {
    const tree = renderer.create(<StatusPage />).toJSON();
    expect(tree).toMatchSnapshot();
  });

});
