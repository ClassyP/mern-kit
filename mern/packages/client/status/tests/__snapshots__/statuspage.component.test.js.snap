// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatusPage Component Test Snapshot test of the StatusPage component 1`] = `
<div
  className="app"
>
  <div
    className="app__header"
    style={
      Object {
        "MozBoxSizing": "border-box",
        "MozTransition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
        "WebkitTapHighlightColor": "rgba(0,0,0,0)",
        "WebkitTransition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
        "backgroundColor": "#00bcd4",
        "borderRadius": "0px",
        "boxShadow": "0 1px 6px rgba(0, 0, 0, 0.12),
               0 1px 4px rgba(0, 0, 0, 0.12)",
        "boxSizing": "border-box",
        "color": "rgba(0, 0, 0, 0.87)",
        "display": "flex",
        "fontFamily": "Roboto, sans-serif",
        "muiPrepared": true,
        "paddingLeft": 24,
        "paddingRight": 24,
        "position": "relative",
        "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
        "width": "100%",
        "zIndex": 1100,
      }
    }
  >
    <h1
      onTouchTap={[Function]}
      style={
        Object {
          "WebkitFlex": "1",
          "boxFlex": 1,
          "color": "#ffffff",
          "flex": "1",
          "fontSize": 24,
          "fontWeight": 400,
          "height": 64,
          "letterSpacing": 0,
          "lineHeight": "64px",
          "margin": 0,
          "msFlex": "1",
          "muiPrepared": true,
          "overflow": "hidden",
          "paddingTop": 0,
          "textOverflow": "ellipsis",
          "whiteSpace": "nowrap",
        }
      }
    >
      Mern-Kit
    </h1>
  </div>
  <div
    className="app__body"
  >
    <div
      className="status__container inner"
    >
      <section
        className="status__title-container"
      >
        <h1
          className="status-code"
        >
          NaN
        </h1>
        <h2
          className="status-message"
        >
          Well I bet you weren't expecting this.
        </h2>
      </section>
      <p
        className="status-details"
      >
        We're sorry, the page you are looking for was not found.  Please  check the url and make sure it is correct.
      </p>
    </div>
  </div>
</div>
`;
